guides/plugins/plugins/content/media/prevent-deletion-of-media-files-referenced-in-your-plugins.md

```diff
@@ -0,0 +1,139 @@
+# Prevent Deletion of Media Files Referenced in your Plugins
+
+{% hint style="info" %}
+The ability to prevent Media entities from being deleted is available since Shopware 6.5.1.0.
+{% endhint %}
+
+## Overview
+
+The Shopware CLI application provides a `media:delete-unused` command which deletes all media entities and their corresponding files which are not used in your application.
+Not used means that it is not referenced by any other entity. This works well in the simple case that all your entity definitions store references to Media entities with correct foreign keys.
+
+However, this does not cover all the possible cases, even for many internal Shopware features. For example the CMS entities store their configuration as JSON blobs with references to Media IDs stored in a nested data structure.
+
+In order to fix the case of Media references that cannot be resolved without knowledge of the specific entity and its features, an extension point is provided via an event.
+
+If you are developing an extension which references Media entities, and you cannot use foreign keys, this guide will detail how to prevent shopware deleting the Media entities your extension references.
+
+## Prerequisites
+
+As most of our plugin guides, this guide was also built upon our [Plugin base guide](../../plugin-base-guide.md).
+Furthermore, you'll have to know about adding classes to the [Dependency injection](../../plugin-fundamentals/dependency-injection.md) container
+and about using a subscriber in order to [Listen to events](../../plugin-fundamentals/listening-to-events.md).
+
+## The deletion process
+
+The `\Shopware\Core\Content\Media\UnusedMediaPurger` service first searches for Media entities that are not referenced by any other entities in the system via foreign keys. Then it dispatches an event containing the Media IDs it believes are unused.
```

Comment:
@JoshuaBehrens for me keeping media files from being deleted by an association is already covered by this sentence. I think adding default folders associations would be more of a guideline

Reply:
Yes, on both counts. All foreign keys are protected automatically, the rest can be protected through the subscribers.

Reply:
> All foreign keys are protected automatically

are they?! Why do I always create a default folder then? :O

Reply:
Ah ok, I am just not on this new feature :D that is why I think I am missing this info.